### PR TITLE
Added support for f303-periph firmware

### PR DESCRIPTION
**Pull request by tridge**
ping @pkocmoud and @jpkh

### CODE REVIEW

File: Tools/AP_Bootloader/mcu_f3.h [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/*
+  support tables for STM32F1
```
Comment (amilcarlucas) [inline Tools/AP_Bootloader/mcu_f3.h]: Should this be STM32F3 ?

File: libraries/AP_FlashStorage/AP_FlashStorage.h [hunk 2]
```diff
@@ -39,7 +39,7 @@
 
 #include <AP_HAL/AP_HAL.h>
 
-#if defined(STM32F1)
+#if defined(STM32F1) || defined(STM32F3)
 /*
   the STM32F1 can't change individual bits from 1 to 0 unless all bits in
```
Comment (amilcarlucas) [inline libraries/AP_FlashStorage/AP_FlashStorage.h]: `the STM32F1 and STM32F3`

File: libraries/AP_HAL_ChibiOS/hwdef/f103-periph/hwdef.dat [hunk 3]
```diff
@@ -57,8 +57,6 @@ PA5 SPI1_SCK SPI1
 PA6 SPI1_MISO SPI1
 PA7 SPI1_MOSI SPI1
 
-SPIDEV rm3100 SPI1 DEVID1 MAG_CS MODE0 1*MHZ 1*MHZ
```
Comment (amilcarlucas) [inline libraries/AP_HAL_ChibiOS/hwdef/f103-periph/hwdef.dat]: Why removing the RM3100 support ? There is no explanation why it was done in the commit message.

File: Tools/AP_Bootloader/mcu_f3.h (hunk 1 above)
Comment (tridge) [reply Tools/AP_Bootloader/mcu_f3.h]: fixed

File: libraries/AP_FlashStorage/AP_FlashStorage.h (hunk 2 above)
Comment (tridge) [reply libraries/AP_FlashStorage/AP_FlashStorage.h]: fixed

File: libraries/AP_HAL_ChibiOS/hwdef/f103-periph/hwdef.dat (hunk 3 above)
Comment (tridge) [reply libraries/AP_HAL_ChibiOS/hwdef/f103-periph/hwdef.dat]: I removed it from the general f103-periph and have now moved it into f103-GPS. We shouldn't be defining this in the generic fw which may not have a mag at all

**Pull request merged**